### PR TITLE
Renable full test-suite

### LINKED ISSUE #7771: Re-enable running whole test suite for binary wheels
### Provide a general description of the issue or problem.

#7767 made it so all our wheels only test the ana tests, due to us now only having one publish block for virtualised aarch64 and regular binary wheels for linux and mac. Once we have actual aarch64 runners or we figure out how to split them back up we should make sure we are running all the tests.

### PR DESCRIPTION
**Pull request by sh-abinash-sh**
## PR Description

1. Used `aarch64` runners explicitly in the publish job to test the full suite for ```manylinux_aarch64 wheels```.
2. I am trying to solve , guidance is really welcomed and very helpful.

Fixes #7771

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -106,6 +106,11 @@ jobs:
           libraries:
             apt:
               - libopenjp2-7
+        - manylinux_aarch64: py312-online
+          posargs: -n auto --dist loadgroup --color=yes
+          libraries:
+            apt:
+              - libopenjp2-7
```
Comment (nabobalis) [inline .github/workflows/ci.yml:113]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -125,6 +130,11 @@ jobs:
           libraries:
             apt:
               - libopenjp2-7
+        - manylinux_aarch64: py313-devdeps
+          posargs: -n auto --dist loadgroup --color=yes
+          libraries:
+            apt:
+              - libopenjp2-7
```
Comment (nabobalis) [inline .github/workflows/ci.yml:137]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -195,14 +205,23 @@ jobs:
       anaconda_package: sunpy
       anaconda_keep_n_latest: 1
       sdist: false
-      test_extras: 'tests-only'
-      test_command: 'pytest -p no:warnings --doctest-rst --pyargs sunpy.io.tests.test_ana'
+      test_extras: 'tests'
```
Comment (nabobalis) [inline .github/workflows/ci.yml:208]:
```suggestion
      test_extras: 'tests-only'
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -195,14 +205,23 @@ jobs:
...
+      test_command: |
+        pytest -p no:warnings --doctest-rst --pyargs sunpy
+        -n auto --dist loadgroup --color=yes
```
Comment (nabobalis) [inline .github/workflows/ci.yml:211]:
```suggestion
      test_command: 'pytest -p no:warnings --doctest-rst --pyargs sunpy'
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -195,14 +205,23 @@ jobs:
...
       submodules: false
       targets: |
         - cp3{10,11,12,13}-macosx_arm64
         - cp3{10,11,12,13}-macosx_x86_64
         - cp3{10,11,12,13}-manylinux_aarch64
```
Comment (nabobalis) [inline .github/workflows/ci.yml:216]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -195,14 +205,23 @@ jobs:
...
         - cp3{10,11,12,13}-manylinux_x86_64
+      aarch64: |
+        manylinux_aarch64
+        pytest -vvv -n auto --dist loadgroup --pyargs sunpy
+        libraries:
+          apt:
+            - libopenjp2-7
+            - graphviz
```
Comment (nabobalis) [inline .github/workflows/ci.yml:224]:
```suggestion
```

File: .github/workflows/ci.yml (hunk 5 above)
Comment (nabobalis) [inline .github/workflows/ci.yml:216]:
```suggestion
        - target: cp3{10,11,12,13}-manylinux_aarch64
          runs-on: arm
```

Comment (nabobalis) [reply .github/workflows/ci.yml:216]: Let us see if this works

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -188,20 +188,21 @@ jobs:
...
     needs: [test, docs]
-    uses: OpenAstronomy/github-actions-workflows/.github/workflows/publish.yml@v1
+    uses: sunpy/github-actions-workflows/.github/workflows/publish.yml@main
     with:
       upload_to_anaconda: ${{ (github.event_name == 'schedule' || github.event_name == 'workflow_dispatch') }}
       anaconda_user: scientific-python-nightly-wheels
       anaconda_package: sunpy
       anaconda_keep_n_latest: 1
       sdist: false
       test_extras: 'tests-only'
```
Comment (Cadair) [inline .github/workflows/ci.yml:198]:
```suggestion
      test_extras: 'tests'
```

**Comment by Cadair** [comment]
Oh fun we are building numcodecs from source, https://github.com/zarr-developers/numcodecs/pull/563 @dstansby you want to finish that PR? :laughing:

**Comment by sh-abinash-sh** [comment]
> numcodecs is causing a significant cost in build time, nearly 5 mins per Python version, however, I think that's fine for the moment.

Ohh should we work on optimizing it?

**Comment by Cadair** [comment]
For numcodecs we really just need to wait on that PR being merged so they provide binaries we can use. The total build time was 28mins and macos was 19, so not the end of the world, definitely worth merging I think.

**Review by Cadair — APPROVED** [review]

**Comment by sh-abinash-sh** [comment]
> For numcodecs we really just need to wait on that PR being merged so they provide binaries we can use. The total build time was 28mins and macos was 19, so not the end of the world, definitely worth merging I think.

That's great!!

**Comment by sh-abinash-sh** [comment]
Sorry mistakenly pressed the close pr button.

**Comment by nabobalis** [comment]
Thanks again @sh-abinash-sh

**Pull request merged**